key_management/src/wallet.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::errors::Error;
+use super::{sigs, KeyInfo, KeyStore};
+use address::Address;
+use crypto::{Signature, SignatureType};
+use std::collections::HashMap;
+use std::str::FromStr;
+
+/// A Key, this contains a key_info, address, and public_key which holds the key type and private key
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct Key {
+    key_info: KeyInfo,
+    // Vec<u8> will be used because Eq has not been implemented for PublicKey type
+    public_key: Vec<u8>,
+    address: Address,
+}
+
+impl Key {
+    /// Return a new Key given key_info
+    pub fn new(key_info: &KeyInfo) -> Result<Key, Error> {
+        let public_key =
+            sigs::to_public(act_sig_type(key_info.key_type()), key_info.private_key())?;
+        let address = sigs::new_address(act_sig_type(key_info.key_type()), public_key.clone())?;
+        Ok(Key {
+            key_info: key_info.clone(),
+            public_key,
+            address,
+        })
+    }
+}
+
+/// This is a Wallet, it contains 2 HashMaps:
+/// - keys which is a HashMap of Keys resolved by their Address
+/// - keystore which is a HashMap of KeyInfos resolved by their Address
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct Wallet {
+    keys: HashMap<Address, Key>,
+    keystore: KeyStore,
+}
+
+impl Wallet {
+    /// Return a new Wallet with a given KeyStore
+    pub fn new(keystore: KeyStore) -> Self {
+        Wallet {
+            keys: HashMap::new(),
+            keystore,
+        }
+    }
+
+    /// Return a new Wallet with Keys contructed by a given list of Keys
+    pub fn new_from_keys(key_vec: Vec<Key>) -> Self {
```

Comment:
Could just have this generic based on an iterator, hard to tell exact use case because there are no usages of this function but no reason to constrain to needing specifically a Vec.

Something like:
```suggestion
    pub fn new_from_keys(key_vec: impl IntoIterator<Item = Key>) -> Self {
```